### PR TITLE
Use SocialActionDrive in Referral Page alpha template

### PR DESCRIPTION
**Pull request by aaronschachter**
### What does this PR do?

This PR refactors the alpha template to render a `SocialActionDrive` component with the user's beta template URL, getting it a whole lot closer to the [design](https://projects.invisionapp.com/share/GYSBOGRSDUA#/screens/372969111):

* Adds optional `shareCardTitle` prop to `SocialActionDrive` for overriding the "Your Online Drive" Card header

* Adds optional `shareCardDescription` prop for intro text required by design

### Any background context you want to provide?

This seemed like a good spot to open for review. There are two biggies left, but shouldn't necessarily block merge as this isn't going live for a bit. 

1 - CSS. The `.social-drive-action` has a width set to 66.6666666666666%, so this looks terrible. Ideally we need to override this on the Referral Page to full width

~2 - Per design, the `Embed` should have title "Check Out Your Referral Page" and a custom description. I doubt we'd want the SEO title/description to read as this, so I've asked about this in our [product team Slack](https://dosomething.slack.com/archives/C09ANFQLA/p1565655219094400) to determine next steps on this item. I was thinking initially that we could pass optional `title` and `description` props to the `Embed`, which would entail another two extra additional props on a `SocialDriveActionContainer` in order to pass these along to the `Embed` within (feels kinda yucky and defeating of an `Embed`)~ Non-issue per https://github.com/DoSomething/phoenix-next/pull/1538#issuecomment-521033247

### What are the relevant tickets/cards?

* https://www.pivotaltracker.com/story/show/167602877
* https://www.pivotaltracker.com/story/show/167633963

### Checklist

* [ ] Documentation added for new features/changed endpoints.
* [ ] Added appropriate feature/unit tests.
* [ ] Added screenshot to PR description of related front-end updates on **small** screens.
* [ ] Added screenshot to PR description of related front-end updates on **medium** screens.
* [ ] Added screenshot to PR description of related front-end updates on **large** screens.

### CODE REVIEW

File: resources/assets/components/pages/ReferralPage/templates/AlphaTemplate.js [hunk 1]
```diff
@@ -1,35 +1,30 @@
...
-        card! The more friends you refer, the more gift cards you earn.
-        (Psst...there’s no limit on how many you can refer!)
-      </p>
-      <h3>
-        <a href={url}>Check Out Your Referral Page</a>
-      </h3>
-      <p>
-        Welcome to {firstName}’s refer a friend page. Do something, get
-        something...
-      </p>
```
Comment (aaronschachter) [inline resources/assets/components/pages/ReferralPage/templates/AlphaTemplate.js]: This `h3` and `p` content is what we want displayed within the `Embed` component in the `SocialDriveActionContainer` (could be passed as more props per PR description)

**Comment by weerd** [comment]
@aaronschachter since this includes some design/front-end visuals, can you include screenshots to the small, medium, large examples in the description?

**Comment by weerd** [comment]
> 1 - CSS. The .social-drive-action has a width set to 66.6666666666666%, so this looks terrible. Ideally we need to override this on the Referral Page to full width

We should use the new CSS Grid definitions set up for Phoenix for this page!

**Comment by aaronschachter** [comment]
> Unless this isn't yet the full design application to this alpha template, then no worries for now

Yes, this isn't the full design yet (per the CSS issue). Will do once each template is complete.

> We should use the new CSS Grid definitions set up for Phoenix for this page!

Where can I find docs on how to use this? Wouldn't we then need to additionally refactor the other campaign components that can display a `SocialActionDrive`, since it's width is[ hardcoded to 66%](https://github.com/DoSomething/phoenix-next/blob/master/resources/assets/components/actions/SocialDriveAction/social-drive.scss#L7)?

**Comment by aaronschachter** [comment]
Re #2 in the PR description, we're going to go with using the standard page title and description to use within the `Embed` block of the `SocialActionDrive` -- so it's a non-issue.

**Comment by aaronschachter** [comment]
If this looks good to merge as is (knowing this isn't user facing and still needs styling), I can merge this and start on cleanup per post-merge suggestions in #1537.

**Comment by weerd** [comment]
> Where can I find docs on how to use this? Wouldn't we then need to additionally refactor the other campaign components that can display a SocialActionDrive, since it's width is hardcoded to 66%?

Ok, lets tackle this in an upcoming PR. We're moving all pages to use CSS Grid (we don't have documentation on it, but have just been referring to existing examples (Story Page, Explore Campaigns page, new Marquee Landing Page template, etc) and the grid-based classes in [`base.scss`](https://github.com/DoSomething/phoenix-next/blob/master/resources/assets/scss/base.scss#L693-L846).

And yep, we need to refactor the `SocialActionDrive` component regardless since the width is defined as `66%` and thus will show up compressed where it's used whether we use CSS Grid or not.

File: resources/assets/components/actions/SocialDriveAction/SocialDriveAction.js [hunk 2]
```diff
@@ -55,13 +55,24 @@ class SocialDriveAction extends React.Component {
...
     const shortenedLink = this.state.shortenedLink;
 
     return (
       <div className="clearfix padding-bottom-lg">
         <div className="social-drive-action">
-          <Card title="Your Online Drive" className="rounded bordered">
+          <Card title={shareCardTitle} className="rounded bordered">
+            {shareCardDescription ? (
+              <div className="padded">
+                <p>{shareCardDescription}</p>
```
Comment (weerd) [inline resources/assets/components/actions/SocialDriveAction/SocialDriveAction.js]: Does the `shareCardDescription` come in as a string or as markdown? I'm guessing a string right? If so, we can simplify this and add the `padded` class name to the `<p>` tag and remove the enclosing `<div>`.

File: resources/assets/components/pages/ReferralPage/templates/AlphaTemplate.js [hunk 3]
```diff
@@ -1,35 +1,30 @@
...
+import SocialDriveActionContainer from '../../../actions/SocialDriveAction/SocialDriveActionContainer';
+
 const AlphaTemplate = props => {
-  const { firstName, primaryCampaignId, userId } = props;
-  // @TODO: This URL should be shortened via Bertly.
-  let url = `/us/join?user_id=${userId}`;
+  const { primaryCampaignId, userId } = props;
+
+  let url = `${PHOENIX_URL}/us/join?user_id=${userId}`;
   if (primaryCampaignId) {
```
Comment (weerd) [inline resources/assets/components/pages/ReferralPage/templates/AlphaTemplate.js]: Add a 👾 between variable def and `if` :)

**Comment by weerd** [comment]
@aaronschachter lets chat some time later regarding the CSS Grid stuffs. Aiming to have as much new work/new pages start utilizing the grid sooner rather than later to help with the transition. :)

**Pull request merged**